### ISSUE TITLE
combined options where the last option has an argument fails

### ISSUE THREAD

**Issue by bmesuere** [ddfreyne/cri#46]
When you combine options, and the last option has an argument, the argument is ignored.

For example, I think the following code should run fine, but results in an error:

``` ruby
    def test_combined_options_with_argument
      input       = %w( foo -ab xxx bar )
      definitions = [
        { :long => 'aaa', :short => 'a', :argument => :forbidden },
        { :long => 'bbb', :short => 'b', :argument => :required },
      ]

      parser = Cri::OptionParser.parse(input, definitions)
      assert(parser.options[:aaa])
      assert_equal({ :bbb => 'xxx' },  parser.options)
      assert_equal(%w(foo bar), parser.arguments)
    end
```

error:

```
Cri::OptionParserTestCase#test_combined_options_with_argument:
Cri::OptionParser::OptionRequiresAnArgumentError: b
    /Users/bart/Code/ruby/cri/lib/cri/option_parser.rb:230:in `block in handle_dash_option'
    /Users/bart/Code/ruby/cri/lib/cri/option_parser.rb:223:in `each'
    /Users/bart/Code/ruby/cri/lib/cri/option_parser.rb:223:in `handle_dash_option'
    /Users/bart/Code/ruby/cri/lib/cri/option_parser.rb:173:in `run'
    /Users/bart/Code/ruby/cri/lib/cri/option_parser.rb:105:in `parse'
    /Users/bart/Code/ruby/cri/test/test_option_parser.rb:289:in `test_combined_options_with_argument'
```

**Comment by ddfreyne** [comment]
Yup, makes sense. I think I never explicitly supported this because it’s not super common to invoke commands like this (at least not in nanoc).

Similarly, when both `b` and `c` have required arguments, the following should work:

```
foo -abc xxx yyy
```

Resulting in the options

```ruby
{
  aaa: true,
  bbb: 'xxx',
  ccc: 'yyy',
}
```

**Comment by bmesuere** [comment]
Your suggestion can't be implemented without introducing ambiguity, for example if `b` and `c` have optional arguments.

**Comment by ddfreyne** [comment]
This is implemented in #56 and will be part of the upcoming Cri release! Sorry for the long delay.

> Your suggestion can't be implemented without introducing ambiguity, for example if `b` and `c` have optional arguments.

The implementation currently takes argument values eagerly, so if `b` and `c` have optional arguments, they’ll get values `xxx` and `yyy`, respectively.

**Comment by ddfreyne** [comment]
This is implemented in Cri 2.9.0, released just now.
